clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -256,11 +244,7 @@ private EhcacheEntityResponse invokeInternal(TimeoutDuration timeLimit, EhcacheE
          * elements.  The server-side exception must be wrapped in a client-side exception
          * to provide proper stack trace for analysis.
          */
-        Exception cause = ((Failure)response).getCause();
-        if (cause instanceof ClusteredEhcacheException) {
-          throw ((ClusteredEhcacheException) cause).withClientStackTrace();
-        }
-        throw new RuntimeException(message + " error: " + cause.toString(), cause);
+        throw ((Failure)response).getCause().withClientStackTrace();
```

Comment:
Why not push that `withClientStackTrace()` into the `Response.decode` method directly?